src/test/java/net/sf/jabref/importer/fileformat/RISImporterTest.java

```diff
@@ -28,82 +27,147 @@ public void setUp() throws Exception {
+        Assert.assertEquals(risImporter.getCLIId(), "ris");
     }
 
     @Test
-    public void testProcessSubSup() {
+    public void testIsRecognizedFormat1() throws IOException {
 
-        HashMap<String, String> hm = new HashMap<>();
-        hm.put("title", "/sub 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_3$", hm.get("title"));
+        RisImporter importer = new RisImporter();
 
-        hm.put("title", "/sub   3   /");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_3$", hm.get("title"));
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest1.ris")) {
+            Assert.assertTrue(importer.isRecognizedFormat(stream));
+        }
 
-        hm.put("title", "/sub 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{31}$", hm.get("title"));
+    }
 
-        hm.put("abstract", "/sub 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_3$", hm.get("abstract"));
+    @Test
+    public void testIsRecognizedFormat2() throws IOException {
+        RisImporter importer = new RisImporter();
 
-        hm.put("review", "/sub 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{31}$", hm.get("review"));
+        try (InputStream stream = RISImporterTest.class.getResourceAsStream("RisImporterTest2.ris")) {
+            Assert.assertFalse(importer.isRecognizedFormat(stream));
+        }
+    }
 
-        hm.put("title", "/sup 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^3$", hm.get("title"));
 
-        hm.put("title", "/sup 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^{31}$", hm.get("title"));
+    @Test
+    public void testImportEntries() throws IOException {
+        RisImporter risImporter = new RisImporter();
+        BibtexImporter bibImporter = new BibtexImporter();
 
-        hm.put("abstract", "/sup 3/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^3$", hm.get("abstract"));
+        try (InputStream risStream = RISImporterTest.class.getResourceAsStream("RisImporterTest1.ris");
+                InputStream bibStream = BibtexImporter.class.getResourceAsStream("RisImporterTest1_comparefile.bib")) {
+            List<BibEntry> risEntries = risImporter.importEntries(risStream, new OutputPrinterToNull());
+            List<BibEntry> bibEntries = bibImporter.importEntries(bibStream, new OutputPrinterToNull());
 
-        hm.put("review", "/sup 31/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$^{31}$", hm.get("review"));
+            Assert.assertEquals(1, risEntries.size());
+            Assert.assertEquals(1, bibEntries.size());
 
-        hm.put("title", "/sub $Hello/");
-        IsiImporter.processSubSup(hm);
-        Assert.assertEquals("$_{\\$Hello}$", hm.get("title"));
+            BibtexEntryAssert.assertEquals(bibEntries, risEntries);
```

Comment:
I would recommend to use data driven tests here as well. See comment https://github.com/JabRef/jabref/pull/508#issuecomment-196202065